community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/index/storage/FailureStorage.java

```diff
@@ -115,6 +115,10 @@ public synchronized void storeIndexFailure( String failure ) throws IOException
         File failureFile = failureFile();
         try ( StoreChannel channel = fs.open( failureFile, "rw" ) )
         {
+            byte[] existingData = new byte[(int) channel.size()];
+            channel.read( ByteBuffer.wrap( existingData ) );
+            channel.position( lengthOf( existingData ) );
```

Comment:
Yes, I think you're right. And better than nothing is at least something. The fundamentals of the design of this class is a bit out of scope for that change tho, isn't it?